pilot/pkg/security/model/authentication.go

```diff
@@ -264,9 +264,9 @@ func ApplyToCommonTLSContext(tlsContext *tls.CommonTlsContext, proxy *model.Prox
 		CaCertificatePath: proxy.Metadata.TLSServerRootCert,
 	}
 
-	// TODO: if subjectAltName ends with *, create a prefix match as well.
 	// TODO: if user explicitly specifies SANs - should we alter his explicit config by adding all spifee aliases?
-	matchSAN := util.StringToExactMatch(subjectAltNames)
+	// If subjectAltName ends with *, create a prefix match as well.
+	matchSAN := util.StringToExactMatchWithFallback(subjectAltNames)
```

Comment:
https://github.com/istio/api/blob/master/networking/v1alpha3/destination_rule.proto#L948 - based on this doc, I think we assume it is exact match. I get the intent of the PR, can  we clarify/update docs before we merge this?

Reply:
Thanks for the comments! Proposing doc updates here: https://github.com/istio/api/pull/1966.

Reply:
Good catch, thanks! Right, should applied to gateway, updated. Also attaching a release note.